### PR TITLE
add Azure SSO

### PR DESCRIPTION
**Pull request by mmjjoozz**

### CODE REVIEW

File: api_reflector/admin.py [hunk 1]
```diff
@@ -2,16 +2,59 @@
...
+class RestrictedAdminView(AdminIndexView):
+    """
+    Overrides default Flask-Admin admin view to implement OSS authentication before accessing.
+    """
+
+    def is_accessible(self):
+        if azure.authorized:
+            try:
+                resp = azure.get("/v1.0/me")
+                assert resp.ok
```
Comment (Mick-Cadac) [inline api_reflector/admin.py:23]: Please change the asserts to a check. Asserts can be optimised out if certain flags are set.

File: api_reflector/admin.py [hunk 2]
```diff
@@ -2,16 +2,59 @@
...
+class RestrictedView(ModelView):
+    """
+    Overrides ModelView to implement OSS authentication before accessing.
+    """
+
+    def is_accessible(self):
+        if azure.authorized:
+            try:
+                resp = azure.get("/v1.0/me")
+                assert resp.ok
```
Comment (Mick-Cadac) [inline api_reflector/admin.py:45]: Please change the asserts to a check. Asserts can be optimised out if certain flags are set.

**Review by backwardspy — CHANGES_REQUESTED** [review]
Looks good so far. I have a few comments after trying it out locally.

**There seems to be no way to disable authentication.**

Requiring every developer to set up Azure SSO before they can use the project is not ideal. My suggestion is to provide an `azure_auth_enabled` setting that is false by default. That setting can then be used to decide whether to restrict the admin views, and register the azure blueprint in the API.

If you add that setting, I would also suggest changing the other azure settings so that they're optional unless the auth is enabled. I think you can achieve this with a [custom validator](https://pydantic-docs.helpmanual.io/usage/validators/) on each setting.

There should be some documentation written in the README to cover how to enable authentication. One other thing I had to change to make authentication work is setting this environment variable in my `.env` file:

```
OAUTHLIB_INSECURE_TRANSPORT=1
```

**Redirect to root after login**

When I was testing, after I logged into my Microsoft account I was redirected back to `/` instead of `/admin/`. It would be nice if the application remembered which URL the user was trying to navigate to, and redirected them there. I believe this is supported in OAuth, so it's likely that `flask-dance` provides functionality for this.

File: settings.py [hunk 3]
```diff
@@ -26,6 +26,9 @@ def validate(cls, value: str) -> str:
 
 class Settings(BaseSettings):
     secret_key: str
+    azure_client_id: str
+    azure_client_secret: str
+    azure_tenant: str
```
Comment (backwardspy) [inline settings.py:31]: Very minor comment, but I'd prefer if these settings were visually separated from `secret_key`. I'd like to keep the settings grouped together with blank lines between each group to hopefully assist people's understanding of what these settings do.

**Review by backwardspy — APPROVED** [review]
lgtm

**Review by Mick-Cadac — APPROVED** [review]

**Pull request merged**